libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -482,6 +482,9 @@ void AP_SerialManager::init()
                     state[i].uart->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
                     break;
 
+                case SerialProtocol_RichenPower:
```

Comment:
I think this should be SerialProtocol_Generator.

Reply:
The protocol is very much RichenPower (H2) specific.  No other generator brand will speak this protocol.

Reply:
.. but the point of the protocol is just to allow the driver to find out which serial port it should use.  For example, have a peek at what we do for rangerfinders or proximity sensors.  Even though different lidar use different serial protocols, we just have one enum entry for all lidar.

Reply:
Duh.  That should have been obvious to me, sorry.

There aren't parameters for this stuff yet, so no `GEN_TYPE`, of course.  In time....

Reply:
Changed.